class.jetpack-user-agent.php

```diff
@@ -289,223 +55,52 @@ function get_mobile_user_agent_name() {
 	 * @return string Name of the platform, false otherwise.
 	 */
 	public function get_platform() {
-		if ( isset( $this->_platform ) ) {
-			  return $this->_platform;
-		}
-
-		if ( strpos( $this->useragent, 'windows phone' ) !== false ) {
-			 $this->_platform = self::PLATFORM_WINDOWS;
-		} elseif ( strpos( $this->useragent, 'windows ce' ) !== false ) {
-			$this->_platform = self::PLATFORM_WINDOWS;
-		} elseif ( strpos( $this->useragent, 'ipad' ) !== false ) {
-			$this->_platform = self::PLATFORM_IPAD;
-		} elseif ( strpos( $this->useragent, 'ipod' ) !== false ) {
-			$this->_platform = self::PLATFORM_IPOD;
-		} elseif ( strpos( $this->useragent, 'iphone' ) !== false ) {
-			$this->_platform = self::PLATFORM_IPHONE;
-		} elseif ( strpos( $this->useragent, 'android' ) !== false ) {
-			if ( $this->is_android_tablet() ) {
-				$this->_platform = self::PLATFORM_ANDROID_TABLET;
-			} else {
-				$this->_platform = self::PLATFORM_ANDROID;
-			}
-		} elseif ( $this->is_kindle_fire() ) {
-			$this->_platform = self::PLATFORM_ANDROID_TABLET;
-		} elseif ( $this->is_blackberry_10() ) {
-			$this->_platform = self::PLATFORM_BLACKBERRY_10;
-		} elseif ( strpos( $this->useragent, 'blackberry' ) !== false ) {
-			$this->_platform = self::PLATFORM_BLACKBERRY;
-		} elseif ( $this->is_blackberry_tablet() ) {
-			$this->_platform = self::PLATFORM_BLACKBERRY;
-		} elseif ( $this->is_symbian_platform() ) {
-			$this->_platform = self::PLATFORM_SYMBIAN;
-		} elseif ( $this->is_symbian_s40_platform() ) {
-			$this->_platform = self::PLATFORM_SYMBIAN_S40;
-		} elseif ( $this->is_J2ME_platform() ) {
-			$this->_platform = self::PLATFORM_J2ME_MIDP;
-		} elseif ( $this->is_firefox_os() ) {
-			$this->_platform = self::PLATFORM_FIREFOX_OS;
-		} else {
-			$this->_platform = false;
-		}
-
-		return $this->_platform;
+		_deprecated_function( __FUNCTION__, 'Jetpack 8.7', '\Automattic\Jetpack\Device_Detection\User_Agent_Info->get_platform from the `automattic/jetpack-device-detection` package' );
```

Comment:
Should all of those use `__METHOD__` instead of `__FUNCTION__`?

Reply:
Absolutely, good catch, thank you!